draft-ietf-httpbis-client-hints.md

```diff
@@ -179,7 +179,17 @@ Above example indicates that the cache key needs to include the Sec-CH-Example a
 
 The request header fields defined in this document, and those that extend it, expose information about the user's environment to enable proactive content negotiation. Such information may reveal new information about the user and implementers ought to consider the following considerations, recommendations, and best practices.
 
-Transmitted Client Hints header fields SHOULD NOT provide new information that is otherwise not available to the application via other means, such as using HTML, CSS, or JavaScript. Further, sending highly granular data, such as image and viewport width may help identify users across multiple requests. Reducing the set of field values that can be expressed, or restricting them to an enumerated range where the advertised value is close but is not an exact representation of the current value, can improve privacy and reduce risk of linkability by ensuring that the same value is sent by multiple users. However, such precautions can still be insufficient for some types of data, especially data that can change over time.
+Features relying on this document to define Client Hints headers SHOULD take into account the following aspects of the information exposed: 
+
+  - Entropy
+    - The feature SHOULD NOT provide new information that is otherwise not available to the application via other means, such as existing request headers, HTML, CSS, or JavaScript.
+    - Exposing highly granular data may help identify users across multiple requests to different origins. Reducing the set of field values that can be expressed, or restricting them to an enumerated range where the advertised value is close but is not an exact representation of the current value, can improve privacy and reduce risk of linkability by ensuring that the same value is sent by multiple users.
+  - Sensitivity
+    - The feature SHOULD NOT expose user sensitive information. To that end, information available to the application, but gated behind specific user actions (e.g. a permission prompt or user activation) SHOULD NOT be exposed as a Client Hint.
+  - Change over time
+    - The feature SHOULD NOT expose user information that changes over time, unless the state change itself is also exposed (e.g. through JavaScript callbacks).
+
+Different features will be positioned in different spots on the spectrum between low-entropy, non-sensitive and static information (e.g. user agent information), and high-entropy, sensitive and dynamic information (e.g. geolocation). User agents SHOULD consider the value provided by a particular feature vs these considerations, and MAY have different policies regarding that tradeoff on a per-feature basis.
```

Comment:
"spectrum" implies linear.  Consider rewording.

Reply:
Would "range" be better?

Reply:
Hmm, I don't think "spectrum" connotes linear; spectrum can be linear and nonlinear.

![image](https://user-images.githubusercontent.com/10652/62313398-86dec400-b445-11e9-825e-d38f5bb92f45.png)

That said, if a different term or turn of phrase is a better fit.. 

Riffing.. 

> Each feature will have different characteristics with respect to key considerations such as: static vs. dynamic information, low vs. high entropy, and user sensitive vs non-sensitive information. For example, user agent version number that is shared by many users is static, low entropy, and does not reveal user sensitive information, whereas geolocation is dynamic, high entropy, and user sensitive. User agents SHOULD consider the value provided by each particular feature on a per-feature basis and MAY have different policies on whether and how the feature is made available.

Warmer?

Reply:
@martinthomson friendly ping! :)

Reply:
SGTM